src/Storefront/Framework/Twig/TwigAppVariable.php

```diff
@@ -110,6 +112,17 @@ public function getDebug(): bool
      */
     public function getFlashes(string|array|null $types = null): array
     {
-        return $this->appVariable->getFlashes($types);
+        $flashes = $this->appVariable->getFlashes($types);
+
+        if ($flashes === []) {
+            self::$displayed_flashes = true;
```

Comment:
this is the simplest thing that might work, however first it is ugly, and second it does not catch when the flashes were read in php stack or over `app.session.flashbag` in twig, not sure if we should built something more complex that checks flashbag state at the beginning of the request and the end 🤔 
Open for suggestions how to cleanly built that